src/isotools/_transcriptome_io.py

```diff
@@ -644,132 +647,132 @@ def _check_chimeric(chimeric):
     return chimeric_dict, non_chimeric
 
 
-def _add_sample_gene(t, g_start, g_end, g_infos, tr_list, chrom, novel_prefix):
+def _add_sample_gene(t, gene_start, gene_end, gene_infos, transcript_list, chrom, novel_prefix):
     '''add new gene to existing gene in chrom - return gene on success and None if no Gene was found.
-    For matching transcripts in gene, transcripts are merged. Coverage, tr TSS/PAS need to be reset.
+    For matching transcripts in gene, transcripts are merged. Coverage, transcript TSS/PAS need to be reset.
     Otherwise, a new transcripts are added. In this case, splice graph and coverage have to be reset.'''
     if chrom not in t.data:
-        for tr in tr_list:
-            tr['annotation'] = (4, {'intergenic': []})
-        return t._add_novel_gene(chrom, g_start, g_end, g_infos['strand'], {'transcripts': tr_list}, novel_prefix)
+        for transcript in transcript_list:
+            transcript['annotation'] = (4, {'intergenic': []})
+        return t._add_novel_gene(chrom, gene_start, gene_end, gene_infos['strand'], {'transcripts': transcript_list}, novel_prefix)
 
     # try matching the gene by id
     try:
-        best_gene = t[g_infos['ID']]
-        if not best_gene.is_annotated or not has_overlap((best_gene.start, best_gene.end), (g_start, g_end)):
+        best_gene = t[gene_infos['ID']]
+        if not best_gene.is_annotated or not has_overlap((best_gene.start, best_gene.end), (gene_start, gene_end)):
             best_gene = None
         elsefound = []  # todo: to annotate as fusion, we would need to check for uncovered splice sites, and wether other genes cover them
     except KeyError:
         best_gene = None
 
     if best_gene is None:
         # find best matching reference gene from t
-        genes_ol_strand = [g for g in t.data[chrom][g_start: g_end] if g.strand == g_infos['strand']]
+        genes_ol_strand = [gene for gene in t.data[chrom][gene_start: gene_end] if gene.strand == gene_infos['strand']]
 
         if not genes_ol_strand:
             covered_splice = 0
         else:
-            sj = sorted(list({(e1[1], e2[0]) for tr in tr_list for e1, e2 in pairwise(tr['exons'])}))
-            splice_sites = np.array([g.ref_segment_graph.find_splice_sites(sj) if g.is_annotated
-                                     else _find_splice_sites(sj, g.transcripts) for g in genes_ol_strand])
+            sj = sorted(list({(exon1[1], exon2[0]) for transcript in transcript_list for exon1, exon2 in pairwise(transcript['exons'])}))
+            splice_sites = np.array([gene.ref_segment_graph.find_splice_sites(sj) if gene.is_annotated
+                                     else _find_splice_sites(sj, gene.transcripts) for gene in genes_ol_strand])
```

Comment:
More `sj` can be expanded to `splice_junctions` here and in function `_find_matching_gene`.